torchmetrics/metric.py

```diff
@@ -104,6 +104,10 @@ def __init__(
         self._persistent: Dict[str, bool] = {}
         self._reductions: Dict[str, Union[str, Callable[[Union[List[Tensor], Tensor]], Tensor], None]] = {}
 
+        # state management
+        self.is_synced = False
+        self._cache: Optional[Dict[str, Union[List[Tensor], Tensor]]] = None
```

Comment:
since this is new, should we start with making `is_synced` protected, especially if we want to refactor the rest of the metrics API? 
```suggestion
        # state management
        self._is_synced = False
        self._cache: Optional[Dict[str, Union[List[Tensor], Tensor]]] = None
```

Reply:
Good with me !